src/adaptive_graph_of_thoughts/domain/services/neo4j_utils.py

```
@@ -1,216 +1,57 @@
-            _driver = None # Ensure driver is None if connection failed
-            raise  # Re-raise the exception to signal connection failure
-        except Exception as e:
-            logger.error(f"An unexpected error occurred while initializing Neo4j driver: {e}")
-            _driver = None # Ensure driver is None on other errors
-            raise
+        except Exception as e:  # pragma: no cover - just log
+            logger.error(f"Failed to connect to Neo4j: {e}")
+            if _driver is not None:
+                _driver.close()
+            _driver = None
+            raise ServiceUnavailable("Could not connect to Neo4j")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)